firmware/src/main.cpp

```diff
@@ -208,7 +212,48 @@ void testModeLoop() {
   }
 }
 
+unsigned long lastPauseTime;
+bool beepPaused = false;
+uint8_t beepCount = 0;
+bool beepState = false;
+void lowVoltageBeep() {
+  if (!beepPaused) {
+    if (!buzzer.isPlaying()) {
+      if (beepState) {
+        buzzer.playFrequency(440, 250, 7);
+      }
+      else {
+        buzzer.playFrequency(220, 250, 7);
+      }
+      beepState = !beepState;
+      beepCount++;
+    }
+
+    if (beepCount > 10) {
+      beepCount = 0;
+      beepPaused = true;
+      lastPauseTime = millis();
+    }
+  }
+  else {
+    if (millis() - lastPauseTime > 3000) {
+      beepPaused = false;
+    }
+  }
+}
+
 void normalModeLoop() {
+  lastBatteryMV = readBatteryMillivolts();
+
+  isLowVoltage = lastBatteryMV < kMinOperatingMV;
+
+  // Shutdown motors if in low voltage mode
+  if (isLowVoltage) {
+    lowVoltageBeep();
+    rPiLink.buffer.leftMotor = 0;
+    rPiLink.buffer.rightMotor = 0;
+  }
```

Comment:
We should probably add something that checks if we have been in low voltage for the last n loops or t time to avoid spurious readings from affecting performance.

Reply:
Yeah good point. I'll add that in